### PR TITLE
feat: allow scaffolder tasks pagination in the frontend

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

Requires #26378, relates to #25856

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder/src/components/ListTasksPage/ListTasksPage.tsx [hunk 1]
```diff
@@ -51,23 +51,29 @@ export interface MyTaskPageProps {
 const ListTaskPageContent = (props: MyTaskPageProps) => {
   const { initiallySelectedFilter = 'owned' } = props;
   const { t } = useTranslationRef(scaffolderTranslationRef);
+  const [limit, setLimit] = useState(5);
+  const [page, setPage] = useState(0);
```
Comment (acierto) [inline plugins/scaffolder/src/components/ListTasksPage/ListTasksPage.tsx:55]: @drodil I wonder if the initial page "0" is better than "1"?

Usually on UI user see the numeration of pages started with 1, not 0.
Won't it lead in the future to do some kind of conversion (N-1) to get proper offset?

Comment (drodil) [reply plugins/scaffolder/src/components/ListTasksPage/ListTasksPage.tsx:55]: Actually no. The MUI table uses zero-based index for the page, see here: https://mui.com/api/table-pagination/

**Review by acierto — APPROVED** [review]

**Comment by drodil** [comment]
Not stale. Backend support was merged. How about this one @acierto ?

File: plugins/scaffolder-react/api-report.md [hunk 2]
```diff
@@ -205,8 +205,13 @@ export interface ScaffolderApi {
...
   listActions(): Promise<ListActionsResponse>;
   // (undocumented)
-  listTasks?(options: { filterByOwnership: 'owned' | 'all' }): Promise<{
+  listTasks?(options: {
+    filterByOwnership: 'owned' | 'all';
+    limit?: number;
+    offset?: number;
+  }): Promise<{
     tasks: ScaffolderTask[];
+    totalTasks: number;
```
Comment (benjdlambert) [inline plugins/scaffolder-react/api-report.md:214]:
```suggestion
    totalTasks?: number;
```

File: plugins/scaffolder-react/src/api/types.ts [hunk 3]
```diff
@@ -215,7 +215,9 @@ export interface ScaffolderApi {
 
   listTasks?(options: {
     filterByOwnership: 'owned' | 'all';
-  }): Promise<{ tasks: ScaffolderTask[] }>;
+    limit?: number;
+    offset?: number;
+  }): Promise<{ tasks: ScaffolderTask[]; totalTasks: number }>;
```
Comment (benjdlambert) [inline plugins/scaffolder-react/src/api/types.ts:220]:
```suggestion
  }): Promise<{ tasks: ScaffolderTask[]; totalTasks?: number }>;
```

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Just a few API changes, need to make them optional like we did in the backend to avoid a breaking change before 2.x

File: plugins/scaffolder/api-report.md [hunk 4]
```diff
@@ -558,8 +558,13 @@ export class ScaffolderClient implements ScaffolderApi_2 {
...
   listActions(): Promise<ListActionsResponse_2>;
   // (undocumented)
-  listTasks(options: { filterByOwnership: 'owned' | 'all' }): Promise<{
+  listTasks(options: {
+    filterByOwnership: 'owned' | 'all';
+    limit?: number;
+    offset?: number;
+  }): Promise<{
     tasks: ScaffolderTask_2[];
+    totalTasks: number;
```
Comment (benjdlambert) [inline plugins/scaffolder/api-report.md:567]:
```suggestion
    totalTasks?: number;
```

**Comment by drodil** [comment]
Thanks @benjdlambert ! Also required a rebase as the api-reports were renamed so now it should be ok :)

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**